docker_cli/_docker_mamba.py

```diff
@@ -0,0 +1,235 @@
+
+    Returns
+    -------
+    str
+        The generated Dockerfile body.
+    """
+    return _mamba_spec_command(
+        specfile=env_specfile, command="install", channels=["conda-forge"]
+    )
+
+
+def mamba_add_packages_dockerfile(
+    packages: Iterable[str], channels: Optional[Iterable[str]]
+) -> str:
+    """
+    Creates a Dockerfile body for adding the given packages to a micromamba environment.
+
+    Parameters
+    ----------
+    packages : Iterable[str]
+        The set of packages to be added to the environment.
+    channels: Iterable[str], optional
+        Channels to look for packages in.
+
+    Returns
+    -------
+    str
+        The Dockerfile body.
+    """
+    return _mamba_spec_command(packages=packages, channels=channels, command="install")
+
+
+def mamba_lockfile_command(
+    env_name: str,
+) -> str:
+    """
+    Returns a command to generate a lockfile with micromamba.
+
+    Parameters
+    ----------
+    env_name : str
+        The name of the environment for which a Dockerfile should be generated.
+
+    Returns
+    -------
+    str
+        The command.
+    """
+    cmd = f"micromamba env export --name {env_name} --explicit --no-md5"
+    return cmd
+
+
+def micromamba_docker_lines():
+    return textwrap.dedent(
+        """
+        ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+        USER $MAMBA_USER
+        """
+    ).strip()
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    packages: Iterable[str],
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    specfile: os.PathLike[str] | str,
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+def _mamba_spec_command(
+    command, *, channels, specfile=None, packages=None, env_name=None
+) -> str:
+    if (specfile is not None) and (packages is not None):
+        raise ValueError("Specfile and packages both given. Use only one.")
+    if command not in ["install", "create"]:
+        raise ValueError(
+            "Improper value given for micromamba specs installation. "
+            'Please use one of "install" or "create".'
+        )
+
+    # If a name was given for the environment, an argument will be added to all
+    # micromamba installation commands.
+    if env_name is None:
+        name_arg: str = ""
+    else:
+        name_arg = f" -n {env_name}"
+
+    # Put together the channels argument
+    if channels is not None:
+        channels_arg = f" -c {shlex.join(channels)} --override-channels"
+
+    # If a specfile is given, give the instructions for copying and installing a
+    # specfile.
+    if specfile is not None:
+        install_command = textwrap.dedent(
+            f"""
+            COPY {specfile} /tmp/spec-file.txt
+            RUN micromamba {command}{name_arg}{channels_arg} -y -f /tmp/spec-file.txt \\
+             && rm /tmp/spec-file.txt \\
```

Comment:
I don't think you want a backslash here since this is the end of the `RUN` command -- we're not continuing onto the next line.

```suggestion
             && rm /tmp/spec-file.txt
```